### ISSUE TITLE
Case when c_in > 1 and soft = True

### ISSUE THREAD

**Issue by jbohnslav** [piergiaj/tgm-icml19#4]
Hi there,

Thanks for the great paper and code. In line 73 of `tgm.py`, you define how you combine stacks of C_in representations: https://github.com/piergiaj/tgm-icml19/blob/340e2ce66439790eb26086d01f773d444fda6ea9/tgm.py#L73

`f` at this point is a tensor of shape `N x C_out x C_in x D x T`. Taking the sum over the first dimension means you sum over the output dimension, resulting in a tensor that's `N x C_in x D x T`.  The softmax has already been taken over the second dimension, which I think is correct. I think it should be: 
`f = torch.sum(a * f, dim=2)`

**Comment by beibeirory** [comment]
I am sorry to disturb you. Could you tell the loss and accuracy during your training process? I can not reproduce the results reported by the author. Thank you. @jbohnslav

**Comment by piergiaj** [comment]
That change seems like it is correct. But I’ll need to closely check the code and tensor shapes to make sure.
